content/building-blocks/automate-and-execute-your-work/automate-your-workflow/ghactions-workflow.md

```diff
@@ -0,0 +1,187 @@
+---
+title: "Build a workflow on a GitHub hosted runner" 
+description: "Learn a practical example of a GitHub actions workflow for code formatting and testing"
+keywords: "GitHub Actions, Data Science, Project, Project Structure, Reproducibility, Automation, Testing, Formatting, Workflows, Runners, Jobs, Events, Testing, Formatting"
+weight: 2
+author: "Fernando Iscar"
+authorlink: "https://www.linkedin.com/in/fernando-iscar/"
+draft: false
+date: 2023-10-14T22:01:14+05:30 
+aliases: 
+  - /github_actions/workflow
+---
+
+## Overview
+
+This building block delves deeper into GitHub Actions by providing a hands-on example. We’ll craft a workflow specifically for code formatting and testing. After that, you'll discover how to integrate and track it into your own repository.
```

Comment:
This may not be strictly necessary but I think that adding a reference to Python code explicitly here may be interesting

"We’ll craft a workflow specifically for Python code formatting and testing. After that, you'll discover how to integrate and track it into your own repository."